### PR TITLE
correct averagepool kernel shape in dilation test case

### PR DESCRIPTION
**Pull request by liqunfu**
### Description
kernel shape is incorrect with test case: test_averagepool_2d_dilations

### Motivation and Context
correct kernel shape

### CODE REVIEW

File: onnx/defs/nn/defs.cc [hunk 1]
```diff
@@ -313,7 +313,7 @@ ONNX_OPERATOR_SET_SCHEMA(
             "AveragePool",
             "average",
             "The output of each pooling window is divided by the number of elements (exclude pad when attribute count_include_pad is zero).",
-            false,
+            true,
```
[the remark this answers is not in the document]
Comment (gramalingam) [reply onnx/defs/nn/defs.cc:316]: Adding a comment what each parameter is will help. E.g., `/*default-value*/`

File: onnx/backend/test/case/node/averagepool.py [hunk 2]
```diff
@@ -471,12 +471,24 @@ def export_averagepool_2d_dilations() -> None:
             "AveragePool",
             inputs=["x"],
             outputs=["y"],
-            kernel_shape=[3, 3],
+            kernel_shape=[2, 2],
             strides=[1, 1],
             dilations=[2, 2],
             ceil_mode=True,
         )
-        x = (np.arange(16) + 1).astype(np.float32).reshape((1, 1, 4, 4))
```
Comment (gramalingam) [inline onnx/backend/test/case/node/averagepool.py:479]: Minor point: Retaining a comment like `# input shape: [1, 1, 4, 4]` will be helpful.

**Review by jcwchen — COMMENTED** [review]
Let's expediate this PR for cherry-pick. Operators.md will need to be updated as well. Thanks!

File: onnx/test/test_backend_reference.py [hunk 3]
```diff
@@ -161,7 +161,9 @@ def run_node(cls, node, inputs, device=None, outputs_info=None, **kwargs):
 
 # The following tests fail due to a shape mismatch.
 backend_test.exclude(
-    "(test_center_crop_pad_crop_axes_hwc_expanded|test_lppool_2d_dilations)"
+    "(test_center_crop_pad_crop_axes_hwc_expanded"
+    "|test_lppool_2d_dilations"
+    "|test_averagepool_2d_dilations)"
```
Comment (xadupre) [inline onnx/test/test_backend_reference.py:166]: The reference evaluation will be fixed in another PR?

Comment (liqunfu) [reply onnx/test/test_backend_reference.py:166]: yes https://github.com/onnx/onnx/issues/5161
this PR is for 1.14 release. Ort release depend on this PR so need to merge and cherry-pick by release sooner the better. Fix of reference implementation can come after.

File: onnx/defs/nn/defs.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (liqunfu) [reply onnx/defs/nn/defs.cc]: comments added for use_dilation and support8bit

**Review by gramalingam — APPROVED** [review]
